### PR TITLE
fix(sqllab): dedupe table_schemas in active_tab

### PR DESCRIPTION
**Pull request by justinpark**
### SUMMARY
_Addition to #23265 fix_
When SQLLAB_BACKEND_PERSISTENCE has been deactivated after activated, active_tab will be provided during localstorage persistence mode.
This causes the appending duplicated table_schemas value every time page is loaded. (This causes the increasing localStorage size)
This commit adds the dedupe logic to clean up the duplicated values.

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF

After:
Superset

Before:
Screenshot_2023-03-27_at_4_31_59_PM

### TESTING INSTRUCTIONS
set SQLLAB_BACKEND_PERSISTENCE on and open SqlLab
select one table to fetch the table schema info
disable SQLLAB_BACKEND_PERSISTENCE and open SqlLab
keep refresh the page and then check the redux value in the local storage

### ADDITIONAL INFORMATION
- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

@michael-s-molina @ktmud

### CODE REVIEW

**Comment by tang350476527** [comment]
谢谢您的来信，您的邮件已收到！

File: superset-frontend/src/SqlLab/reducers/getInitialState.js [hunk 1]
```diff
@@ -183,9 +186,15 @@ export default function getInitialState({
           },
         };
       });
-      sqlLab.tables.forEach(table =>
-        tables.push({ ...table, inLocalStorage: true }),
-      );
+      sqlLab.tables.forEach(table => {
```
Comment (Antonio-RiveroMartnez) [inline superset-frontend/src/SqlLab/reducers/getInitialState.js:189]: Nit, could this be changed to use `reduce` so it's clear what the intention is while iterating over the array and manipulating the new type?

Comment (justinpark) [reply superset-frontend/src/SqlLab/reducers/getInitialState.js:189]: good call. I will update this part

**Review by Antonio-RiveroMartnez — APPROVED** [review]
LGTM!

**Pull request merged**